imod/mf6/wel.py

```diff
@@ -429,19 +434,21 @@ def to_mf6_pkg(
 
         wells_df = self._create_wells_df()
         nwells_df = len(wells_df["id"].unique())
-        wells_assigned = self._assign_wells_to_layers(wells_df, active, top, bottom, k)
-
-        nwells_assigned = (
-            0 if wells_assigned.empty else len(wells_assigned["id"].unique())
-        )
-
         if nwells_df == 0:
-            raise ValueError("No wells were assigned in package. None were present.")
+            raise ValidationError(
+                "No wells were assigned in package. None were present."
+            )
 
-        if not is_partitioned and nwells_df != nwells_assigned:
-            raise ValueError(
-                "One or more well(s) are completely invalid due to minimum conductivity and thickness constraints."
+        wells_assigned = self._assign_wells_to_layers(wells_df, active, top, bottom, k)
+        filtered_ids_in_assign = self.gather_filtered_well_ids(wells_assigned, wells_df)
+        message_assign = self.to_mf6_package_information(
+            filtered_ids_in_assign, reason_text="permeability/thickness constraints"
+        )
+        if error_on_well_removal and len(filtered_ids_in_assign) > 0:
+            logger.log(
+                loglevel=LogLevel.ERROR, message=message_assign, additional_depth=2
```

Comment:
Why did you add `additional_depth=2`? I don't think it is necessary here. 

The `additional_depth` parameter is used when you want to log the file and line level of one level lower then the current line. This comes in handy for the logging decorator because you don't want to the log the line in the decrator but the line in the method that is decorated

Reply:
Woops, relict of some copy pasta action (I didn't know what the ``additional_depth`` parameter did.) 

Fixed!